tests/openapi/test_strictmode.py

```diff
@@ -0,0 +1,343 @@
+                                                     'or use a different filter.')
+    assert not search_fail.ok
+
+    request_with_validation(
+        api='/collections/{collection_name}/index',
+        method="PUT",
+        path_params={'collection_name': collection_name},
+        query_params={'wait': 'true'},
+        body={
+            "field_name": "city",
+            "field_schema": "keyword"
+        }
+    ).raise_for_status()
+
+    # We created an index on this field so it should work now
+    search_request_with_filter().raise_for_status()
+
+
+def test_strict_mode_unindexed_filter_write_validation():
+    def update_request_with_filter():
+        return request_with_validation(
+            api='/collections/{collection_name}/points/delete',
+            method="POST",
+            path_params={'collection_name': collection_name},
+            query_params={'wait': 'true'},
+            body={
+                "filter": {
+                    "must": [
+                        {
+                            "key": "city",
+                            "match": {
+                                "value": "Berlin"
+                            }
+                        }
+                    ]
+                }
+            })
+
+    update_request_with_filter().raise_for_status()
+
+    # Reset any changes
+    basic_collection_setup(collection_name=collection_name)
+
+    set_strict_mode({
+        "enabled": True,
+        "unindexed_filtering_update": False,
+    })
+
+    search_fail = update_request_with_filter()
+
+    assert search_fail.json()['status']['error'] == (
+        'Forbidden: Index required but not found for "city" of one of the '
+        'following types: [keyword]. Help: Create an index for this key or '
+        'use a different filter.')
+    assert not search_fail.ok
+
+    request_with_validation(
+        api='/collections/{collection_name}/index',
+        method="PUT",
+        path_params={'collection_name': collection_name},
+        query_params={'wait': 'true'},
+        body={
+            "field_name": "city",
+            "field_schema": "keyword"
+        }
+    ).raise_for_status()
+
+    # We created an index on this field so it should work now
+    update_request_with_filter().raise_for_status()
+
+
+def test_strict_mode_max_ef_hnsw_validation():
+    def search_request():
+        return request_with_validation(
+            api='/collections/{collection_name}/points/search',
+            method="POST",
+            path_params={'collection_name': collection_name},
+            body={
+                "vector": [0.2, 0.1, 0.9, 0.7],
+                "limit": 4,
+                "params": {
+                    "hnsw_ef": 5,
+                }
+            }
+        )
+
+    search_request().raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "search_max_hnsw_ef": 3,
+    })
```

Comment:
Note: here we set it to 3 but I think 4 should already fail (untested):

```suggestion
    set_strict_mode({
        "enabled": True,
        "search_max_hnsw_ef": 5,
    })
    
    search_request().raise_for_status()
    
    set_strict_mode({
        "search_max_hnsw_ef": 4,
    })
```